File: deepmind/part_two/reverse-string-in-place.ts
// Reverse array of characters, `xs`, mutatively.
function reverse(xs: Array<string>) {
  let i: number = 0;
  let j: number = xs.length - 1;

  while (i < j) {
    let tmp = xs[i];
    xs[i] = xs[j]
    xs[j] = tmp
    i += 1
    j -= 1
  }
}

File: scratch/facebook/reverse-string-in-place.py
# reverse :: [Char] -> ()
def reverse(xs):
    i = 0
    j = len(xs) - 1
    while i < j:
        xs[i], xs[j] = xs[j], xs[i]
        i += 1
        j -= 1

xs = [list("testing"), list("a"), list("to")]
for x in xs:
    print(x)
    reverse(x)
    print(x)
